File: 프로그래머스/1/12906. 같은 숫자는 싫어/같은 숫자는 싫어.cpp
#include <vector>
#include <iostream>
#include <stack>
#include <algorithm>

using namespace std;

vector<int> solution(vector<int> arr) 
{
    stack<int> st;
    vector<int> result;
    for(int n : arr)
        st.push(n);

    int prev = st.top();
    result.push_back(prev);
    
    while(!st.empty()) {
        // 중복인 경우
        if(prev == st.top())
            st.pop();
        else {
            prev = st.top();
            st.pop();
            result.push_back(prev);
        }
    }
    
    reverse(result.begin(), result.end());
    
    return result;
    
}

File: 프로그래머스/1/12906. 같은 숫자는 싫어/같은 숫자는 싫어.java
import java.util.*;

public class Solution {
    public int[] solution(int []arr) {
        
        Stack<Integer> stack = new Stack<>();
        for(int n : arr){
            if(!stack.isEmpty() && stack.peek() == n){
                continue;
            }
            stack.push(n);
        }
        
        int[] answer = new int[stack.size()];
        for(int j = answer.length - 1 ; j >= 0 ; j--){
            answer[j] = stack.pop();
        }

        return answer;
    }
}